### PR TITLE
expose afterChange and beforeChange in Carousel API

### PR DESCRIPTION
**Pull request by zeevgg**
### 🔦 Summary
expose afterChange and beforeChange in Carousel API

### ✅ Checklist
- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [ ] Story
  - [ X] API description
  - [ ] Other (explain)
- 🔬 Change is tested
  - [ X] Component
  - [ ] Visual test

### CODE REVIEW

File: src/Carousel/Carousel.spec.js [hunk 1]
```diff
@@ -86,6 +87,28 @@ describe('Carousel', () => {
         expect(driver.getCurrentImageIndex()).toBe(0);
       }, 0);
     });
+
+    it('should trigger beforeChange afterChange when switching to the next image', () => {
+      const afterChange = sinon.spy();
```
Comment (shlomitc) [inline src/Carousel/Carousel.spec.js]: Use `jest.fn()` and not `sinon`

File: src/Carousel/Carousel.js [hunk 2]
```diff
@@ -46,18 +46,23 @@ class Carousel extends React.Component {
...
+    /** Index change callback. `(oldIndex, newIndex) => ...` */
+    beforeChange: PropTypes.func,
   };
 
   static defaultProps = {
     infinite: true,
     initialSlideIndex: 0,
     images: [],
     buttonSkin: 'standard',
+    afterChange: null,
```
Comment (shlomitc) [inline src/Carousel/Carousel.js]: I don't think a default prop is needed here

File: src/Carousel/Carousel.spec.js [hunk 3]
```diff
@@ -86,6 +87,28 @@ describe('Carousel', () => {
         expect(driver.getCurrentImageIndex()).toBe(0);
       }, 0);
     });
+
+    it('should trigger beforeChange afterChange when switching to the next image', () => {
```
Comment (nitayneeman) [inline src/Carousel/Carousel.spec.js]: Missing "and" word?

File: src/Carousel/Carousel.spec.js (hunk 1 above)
Comment (zeevgg) [reply src/Carousel/Carousel.spec.js]: ok, not sure why. sinon is used a lot in the project

**Pull request merged**